version: '3'

services:
  web:
    build: .
    command: python manage.py runserver 0.0.0.0:8000
    ports:
      - "8000:8000"
    volumes:
      - .:/code
    environment:
        - DEBUG=1
        - DJANGO_ALLOWED_HOSTS=localhost 127.0.0.1 [::1]
        - CELERY_BROKER=redis://redis:6379/0
        - CELERY_BACKEND=redis://redis:6379/0
    depends_on:
      - db
      - redis
    networks:
      - seotool-network


  celery:
      build: .
      command: celery -A django_project worker -l INFO
      volumes:
          - .:/usr/src/app
      environment:
          - DEBUG=1
          - DJANGO_ALLOWED_HOSTS=localhost 127.0.0.1 [::1]
          - CELERY_BROKER=redis://redis:6379/0
          - CELERY_BACKEND=redis://redis:6379/0
      depends_on:
          - web
          - redis  
      networks:
      - seotool-network


  db:
    image: postgres
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: django_project
    networks:
      - seotool-network
  
  redis:
    image: "redis:alpine"
    networks:
      - seotool-network
    ports:
      - "6379:6379"

  typesense:
    container_name: typesense
    image: typesense/typesense:0.25.1
    ports:
      - target: 8108
        published: 8108
        mode: host
    volumes:
      - vector-data:/data-seo
    command: '--data-dir /data-seo --api-key=xyz --enable-cors'
    networks:
      - seotool-network

networks:
  seotool-network:

volumes:
  pgdata:
  vector-data:
